### PR TITLE
Convert Library Version html to doxygen

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: doxygen/dox/TechnicalNotes.dox [hunk 1]
```diff
@@ -26,7 +26,264 @@
 
 /** \page RELVERSION HDF5 Library Release Version Numbers
 
-\htmlinclude LibraryReleaseVersionNumbers.html
```
Comment (bmribler) [inline doxygen/dox/TechnicalNotes.dox:29]: Shouldn't this be in a separate dox file?

Comment (byrnHDF) [reply doxygen/dox/TechnicalNotes.dox:29]: It was short enough to be here.

File: doxygen/dox/RelVersion.dox [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+</td>
+</tr>
+</table>
+
+The <b>trunk</b> is the center of the system. New features are
+implemented in <b>feature branches</b> and aggregated in the trunk.
+<b>Release branches</b> are then created from the trunk.
+
+Before the 2.0 version, the minor version number of the trunk is always an odd number. From
+the time of Release 1.8.0 to the first 1.10 release, the trunk will be
```
Comment (mattjala) [inline doxygen/dox/RelVersion.dox]: `the trunk will be` -> `the trunk was`

**Review by mattjala — DISMISSED** [review]

File: doxygen/dox/RelVersion.dox (hunk 2 above)
Comment (byrnHDF) [reply doxygen/dox/RelVersion.dox]: Changed

**Review by lrknox — APPROVED** [review]
The text will need more thoughtful review before release.
1. I don't think the major/minor/release difference between the  1.x releases and the 2.x releases is sufficiently clear.
2. I think we changed H5check behavior at some point.  We should verify that the sections around line 170 accurately describe current behavior.

**Review by mattjala — APPROVED** [review]

**Pull request merged**